src/Lazy/uniqBy.ts

```diff
@@ -92,14 +92,14 @@ function uniqBy<A extends Iterable<unknown> | AsyncIterable<unknown>, B>(
 
   if (isIterable(iterable)) {
     return pipe(
-      iterable as Iterable<IterableInfer<A>>,
+      iterable,
       filter((a) => pipe1(f(a), checkAndAdd)),
```

Comment:
I change this line to remove type error.
It was caused by following definition.

```ts
function filter<
  A extends Iterable<unknown> | AsyncIterable<unknown>,
  B extends IterableInfer<A>,
>(
  f: (a: IterableInfer<A>) => a is B,
): (iterable: A) => ReturnIterableIteratorType<A, B>;
```

I can not sure it is right definition.